pkg/kubernetes/deploy.js

```diff
@@ -0,0 +1,464 @@
+            $('#namespace_value input[type=text]').val(appdeployer.namespacetxt);
+            if(ns_selector.val() === '') {
+                if (appdeployer.namespacetxt.trim() === '' || appdeployer.namespacetxt.trim() === 'Custom Namespace' || 
+                        !/^[a-z0-9]+$/i.test(appdeployer.namespacetxt.trim())) {
+                    $('#deploy-app-namespace-field-note').addClass('has-error').show();
+                    valid_ns = false;
+                    display_deploy_button();
+                    return;
+                } else {
+                    valid_ns = true;
+                    display_deploy_button();
+                }
+            } else {
+                var nsv = ns_selector.val();
+                if (nsv.trim() === '' || nsv.trim() === 'Custom Namespace' || !/^[a-z0-9]+$/i.test(nsv.trim())) {
+                    $('#deploy-app-namespace-field-note').addClass('has-error').show();
+                    valid_ns = false;
+                    display_deploy_button();
+                    return;
+                } else {
+                    valid_ns = true;
+                    display_deploy_button();
+                }
+                appdeployer.namespacetxt = ns_selector.val();
+                $('#namespace_value input[type=text]').val(appdeployer.namespacetxt);
+            }
+        });
+
+        manifest_file.on('change', function() {
+            deploy_dialog_remove_file_errors();
+            remove_notifications();
+            $('#namespace_value input[type=text]').val(appdeployer.namespacetxt);
+
+
+            var files, file, reader;
+            files = manifest_file[0].files;
+            if (files.length != 1) {
+                text = _("No metadata file was selected. Please select a Kubernetes metadata file. ");
+                disable_deploy_button();
+                manifest_file_details.show().text(text);
+                manifest_file_note.addClass('has-error').show();
+                valid_manifest = false;
+                display_deploy_button();
+
+                manifest_file_btn.text(_("Select Manifest File..."));
+                return;
+            }
+            file = files[0];
+            if (!file.type.match("json.*")) {
+                text = _("The selected file is not a valid Kubernetes application manifest. ");
+                disable_deploy_button();
+                manifest_file_details.show().text(text);
+                manifest_file_note.addClass('has-error').show();
+
+                valid_manifest = false;
+                display_deploy_button();
+                manifest_file_btn.text(file.name).removeClass('manifest_file_default').addClass('manifest_file');
+                return;
+            }
+            reader = new window.FileReader();
+            reader.onerror = function(event) {
+                text = _("Unable to read the Kubernetes application manifest file.Code " + event.target.error.code + "");
```

Comment:
This is not translatable. We need to use cockpit.format() here. In addition there should be a space before the word "Code".